src/v/iceberg/rest_client/tests/retry_policy_tests.cc

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "iceberg/rest_client/retry_policy.h"
+
+#include <gtest/gtest.h>
+
+namespace r = iceberg::rest_client;
+using enum boost::beast::http::status;
+
+namespace {
+template<typename Ex>
+r::failure throw_and_catch(Ex ex) {
+    try {
+        throw ex;
+    } catch (...) {
+        return r::default_retry_policy{}.should_retry(std::current_exception());
+    }
+}
+} // namespace
+
+TEST(default_retry_policy, status_ok) {
+    r::default_retry_policy p;
+    auto result = p.should_retry(
+      http::downloaded_response{.status = ok, .body = iobuf::from("success")});
+    ASSERT_TRUE(result.has_value());
+    ASSERT_EQ(result.value().body, iobuf::from("success"));
```

Comment:
Can be combined into `ASSERT_THAT(Optional(iobuf::from("success")))`

Reply:
I tried this but it seems to try to copy the iobuf